crmd/lrm.c

```diff
@@ -1808,9 +1859,31 @@ construct_op(lrm_state_t * lrm_state, xmlNode * rsc_op, const char *rsc_id, cons
     op->timeout = crm_parse_int(op_timeout, "0");
     op->start_delay = crm_parse_int(op_delay, "0");
 
+    if (safe_str_eq(op->op_type, CRMD_ACTION_STATUS) && op->interval == 0 && rsc) {
+        lrm_state_update_rsc_metadata(lrm_state, rsc, NULL);
+    } else if (!is_remote_lrmd_ra(NULL, NULL, rsc_id) && rsc) {
+        xmlNode *metadata = lrm_state_get_rsc_metadata(lrm_state, rsc);
+
+        if (metadata) {
+            const char *ra_version = get_ra_version(rsc, metadata);
+            xmlNode *versioned_attrs =  NULL;
+            GHashTable *hash = NULL;
+            char *key = NULL;
+            char *value = NULL;
+            GHashTableIter iter;
+
+            versioned_attrs = first_named_child(rsc_op, XML_TAG_RSC_VER_ATTRS);
+            hash = pe_unpack_versioned_parameters(versioned_attrs, ra_version);
+            g_hash_table_iter_init(&iter, hash);
+            while (g_hash_table_iter_next(&iter, (gpointer *) &key, (gpointer *) &value)) {
+                g_hash_table_replace(params, strdup(key), strdup(value));
+            }
+            g_hash_table_destroy(hash);
```

Comment:
We actually need to destroy the hash table since some memory is allocated for it. But we still can avoid unnecessary copying.